### PR TITLE
Deter leakage from Shamir.

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

**Comment by LLBennett** [comment]
LGTM.

**Comment by LLBennett** [comment]
I see the _subtle_ change. LGTM.

**Comment by jefferai** [comment]
I see what you did there ^

File: shamir/shamir.go [hunk 1]
```diff
@@ -80,30 +81,59 @@ func interpolatePolynomial(x_samples, y_samples []uint8, x uint8) uint8 {
...
-	return expTable[diff]
+
+	ret := expTable[diff]
+
+	// Ensure we return zero if a is zero but aren't subject to timing attacks
+	goodVal = ret
+
+	if subtle.ConstantTimeByteEq(a, 0) == 1 {
+		ret = zero
+	} else {
```
Comment (vishalnayak) [inline shamir/shamir.go]: Do we need this `else` block?

Comment (jefferai) [inline shamir/shamir.go]: Yeppers, to keep timing constant we need a constant number of assignments.

**Pull request merged**